ctru-rs/src/services/cam.rs

```diff
@@ -0,0 +1,1020 @@
+use crate::services::gspgpu::FramebufferFormat;
+use bitflags::bitflags;
+use ctru_sys::Handle;
+use std::cell::RefCell;
+
+pub struct Cam {
+    pub inner_cam: RefCell<InwardCam>,
+    pub outer_right_cam: RefCell<OutwardRightCam>,
+    pub outer_left_cam: RefCell<OutwardLeftCam>,
+    pub both_outer_cams: RefCell<BothOutwardCam>,
+}
+
+bitflags! {
+    #[derive(Default)]
+    struct CamPort: u32 {
+        const NONE = ctru_sys::PORT_NONE;
+        const CAM1 = ctru_sys::PORT_CAM1;
+        const CAM2 = ctru_sys::PORT_CAM2;
+        const BOTH = ctru_sys::PORT_BOTH;
+    }
+}
+
+bitflags! {
+    #[derive(Default)]
+    struct CamSelect: u32 {
+        const NONE      = ctru_sys::SELECT_NONE;
+        const OUT1      = ctru_sys::SELECT_OUT1;
+        const IN1       = ctru_sys::SELECT_IN1;
+        const OUT2      = ctru_sys::SELECT_OUT2;
+        const OUT1_OUT2 = ctru_sys::SELECT_OUT1_OUT2;
+    }
+}
+
+bitflags! {
+    #[derive(Default)]
+    pub struct CamContext: u32 {
```

Comment:
The `CamContext` (aka `CAMU_Context` in the raw bindings) seems a difficult to implement/useless addition. As far as I understand, a context is simply a set of "options" which you can define for the camera environment. After setting the options for each context (of which there are only 2), you can switch between them at will, without having to mess with the options again. It's cool as a concept, but seeing how it has to be used for each cam and etc., it quite easily gives problems with the camera's internal state.

I'd suggest to avoid its use for now, and de-implement it. We can add it back once the API gets more stable. This also counts for the functions that require it.

Reply:
Would it be ok to instead force the use of `CamContext::A` since that's the default when the service is initialized? Then we don't have to remove the functions, just remove the context from the signatures and settings can still be used.

Reply:
> Would it be ok to instead force the use of `CamContext::A` since that's the default when the service is initialized? Then we don't have to remove the functions, just remove the context from the signatures and settings can still be used.

That’s what I was thinking.